scripts/west_commands/packages.py

```diff
@@ -0,0 +1,138 @@
+        parser = parser_adder.add_parser(
+            self.name,
+            help=self.help,
+            description=self.description,
+            formatter_class=argparse.RawDescriptionHelpFormatter,
+            epilog=textwrap.dedent(
+                """
+            Listing packages:
+
+                Run 'west packages <manager>' to list packages for Zephyr and active modules
+                for a given package manager.
+                See 'west packages <manager> --help' for details.
+            """
+            ),
+        )
+
+        parser.add_argument(
+            "-p",
+            "--project",
+            action="append",
+            default=[],
+            dest="projects",
+            help="Project to run the 'packages' command for. "
+            "Use 'zephyr' if the 'packages' command should run for Zephyr itself. "
+            "Option can be passed multiple times. "
+            "If this option is not given, the 'packages' command will run for Zephyr "
+            "and all active modules.",
+        )
+
+        subparsers_gen = parser.add_subparsers(
+            metavar="<manager>",
+            dest="manager",
+            help="select a manager.",
+            required=True,
+        )
+
+        pip_parser = subparsers_gen.add_parser(
+            "pip",
+            help="manage pip packages",
+            formatter_class=argparse.RawDescriptionHelpFormatter,
+            epilog=textwrap.dedent(
+                """
+            Manage pip packages:
+
+                Run 'west packages pip' to print all pip packages or requirements
+                files needed by Zephyr and the active projects.
+
+                The output is compatible with the requirements file format itself.
+            """
+            ),
+        )
+
+        pip_parser.add_argument(
+            "--install",
+            action="store_true",
+            help="Install pip packages instead of listing them. "
+            "A single 'pip install' command is built and executed, passing all arguments "
+            "from the original 'west packages pip --install' command. For example pass "
+            "'--dry-run' to not actually install anything, but print what would be.",
+        )
+
+        return parser
+
+    def do_run(self, args, unknown):
+        if args.manager == "pip":
+            return self.do_run_pip(args, unknown)
+
+        # Unreachable but print an error message if an implementation is missing.
+        self.die(f'Unsupported package manager: "{args.manager}"')
+
+    def do_run_pip(self, args, user_args):
+        packages = []
+        requirements = []
+
+        if not args.projects or "zephyr" in args.projects:
+            requirements.append(ZEPHYR_BASE / "scripts/requirements.txt")
+
+        for module in zephyr_module.parse_modules(ZEPHYR_BASE, self.manifest):
+            module_name = module.meta.get("name", None)
+            if args.projects and module_name not in args.projects:
```

Comment:
I think this is covered with:

```python
        if not args.projects or "zephyr" in args.projects:
            requirements.append(ZEPHYR_BASE / "scripts/requirements.txt")
```

Not mentioning the "manifest" project, but `zephyr` explicitly.

Reply:
> I think this is covered with:

apparently you only read the last sentence related to the **note**, and not everything above which is actually where you have an issue.

I tried to tell you that the module name is not good to use because it might be different from the project name.
For example add the following snippet to `zephyr/module.yml` for the [TI hal](https://github.com/zephyrproject-rtos/hal_ti) together with a `scripts/requirements.txt`.
```
package-managers:
  pip:
    requirements:
      - 'scripts/requirements.txt'
```

The west project name is `hal_ti`, the Zephyr module name is `ti`, as can be seen with:
```
$ west list ti
FATAL ERROR: unknown project name/path: ti
  Hint: use "west list" to list all projects.
```
and:
```
$ west list hal_ti
hal_ti       modules/hal/ti               2e7b95ad079e9f636884eedc6853e6ad98b85f65 https://github.com/zephyrproject-rtos/hal_ti
```

but with `west packages`:
```
$ west packages --project hal_ti pip --install -- --dry-run
ERROR: You must give at least one requirement to install (see "pip help install")
```
because you need to give the module name:
```
$ west packages --project ti pip --install -- --dry-run
Requirement already satisfied: Jinja2 in /home/tora/zephyr-venv/lib/python3.10/site-packages (from -r /projects/github/ncs/modules/hal/ti/scripts/requirements.txt (line 1)) (3.1.4)
```

and when you fix this, you must remember to still be careful on the Zephyr project :wink:

Reply:
Right, misunderstood your comment, will take a look.